### ISSUE TITLE
Objects appear entirely black except for night light textures on PopOS

### ISSUE THREAD

**Issue by C1ffisme** [cosmonium/cosmonium#36]
OS: PopOS 22.04 LTS
Version: 0.3.0 Unstable
CPU: Intel Core i7 6th gen
Graphics: NVIDIA Corporation GM107GLM [Quadro M1200 Mobile] / Mesa Intel® HD Graphics 530 (SKL GT2)

I don't remember the eclipse being this widespread.

Screenshots:

![Screenshot from 2024-04-08 16-16-22](https://github.com/cosmonium/cosmonium/assets/9893266/c3cf2db9-b8b4-4fad-9d8d-a2cabceb08c8)
![Screenshot from 2024-04-08 16-16-56](https://github.com/cosmonium/cosmonium/assets/9893266/a4d40c85-a88c-4a4d-aaf3-700aff8c2f46)
![Screenshot from 2024-04-08 16-17-18](https://github.com/cosmonium/cosmonium/assets/9893266/40e37725-09be-4a94-b993-7f46007b9174)

Weirdly no logs were generated by the 0.3.0 version of the program in `/.local/share/cosmonium/output.log`, until I ran the older 0.2.0 version, which crashed before it would pop me into the simulator at all. Here are the logs from that version if it helps:

```
Using C++ Engine
Loading config file /home/c1ff/.config/cosmonium/config.yaml
Found languages en_US.UTF-8
Cosmonium version: V0.2.1
Python version: 3.7.6
Panda version: 1.10.7 (fa210aec88aa3b16ef3b423573542ed7ee73c0e4) by cosmonium (Dec 22 2020 21:14:35)
Panda Systems:
	 Bullet
	 Freetype
	 OpenSSL
	 eigen
	 libjpeg
	 libpng
	 libtiff
	 nativenet
	 net
	 openexr
	 pipelining
	 stdfloat-double
	 system
	 threads
	 zlib
Data type: double
Panda config file /home/c1ff/.config/cosmonium/config.prc
:audio(error):   load_dso(libp3openal_audio.so) failed, will use NullAudioManager
:audio(error):     No error.
Known pipe types:
  glxGraphicsPipe
(all display modules loaded.)
ATTENTION: default value of option vblank_mode overridden by environment.
Using floating point buffer
Hardware Vendor: Intel
Driver Renderer: Mesa Intel(R) HD Graphics 530 (SKL GT2) (4.6 (Compatibility Profile) Mesa 24.0.0-1pop0~1706872735~22.04~0fa430c)
Shader version: 460
Hardware sRGB: True
sRGB textures: True
Hardware instancing: True
Hardware tessellation: True
Inverse-Z buffer: False
HDR: False
sRGB: True
Render to buffer: True
Floating point buffer: True
Texture array: True
Cache directory: /home/c1ff/.cache/cosmonium
Planes:  1.0 inf
Found languages en_US.UTF-8
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/defaults.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/cosmonium.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/solar-system.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/frames.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/ssd.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/manual-orbits.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/sun/sun.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/mercury/mercury.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/venus/venus.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/earth/earth.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/earth/moon/moon.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/earth/hst/hst.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/earth/iss/iss.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/mars/mars.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/jupiter/jupiter.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/jupiter/inner-moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/jupiter/moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/jupiter/irregular-moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/saturn/saturn.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/saturn/inner-moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/saturn/moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/saturn/irregular-moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/uranus/uranus.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/uranus/inner-moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/uranus/moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/uranus/irregular-moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/neptune/neptune.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/neptune/inner-moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/neptune/moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/neptune/irregular-moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/dwarfs/dwarfs.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/dwarfs/pluto/pluto.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/dwarfs/pluto/moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/dwarfs/ceres/ceres.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/dwarfs/eris/eris.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/dwarfs/haumea/haumea.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/dwarfs/makemake/makemake.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/asteroids/asteroids.yaml (cached)
DB rotations : Element astraea not found
DB rotations : Element 6-hebe not found
DB rotations : Element 7-iris not found
DB rotations : Element 8-flora not found
DB rotations : Element 9-metis not found
DB rotations : Element 10-hygiea not found
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/tnos/tnos.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/comets/comets.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/solar-system/interstellars/interstellars.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/stars.yaml (cached)
Unknown class '(K)'
Unknown class 'esdK7'
Unknown class 'esdM1.5'
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/galaxies.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/procedural/procedural.yaml (cached)
Warning: 'noise' entry is deprecated, use 'func' instead'
Warning: 'noise' entry is deprecated, use 'func' instead'
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/constellations.yaml (cached)
Loading /home/c1ff/Applications/cosmonium-old/cosmonium/data/asterisms.yaml (cached)
Could not find star ALF CRU
Creating octree...
Creation time: 0.07278633117675781
:task(error): Exception occurred in PythonTask async_startCallbackTask
WARNING: Token 'BOOL' defined, but not used
WARNING: Token 'FLOAT' defined, but not used
WARNING: Token 'INT' defined, but not used
WARNING: Token 'NAME' defined, but not used
WARNING: There are 4 unused tokens
Traceback (most recent call last):
  File "cosmonium.workers", line 66, in callbackTask
  File "cosmonium.cosmonium", line 462, in configure_scene
  File "cosmonium.ui.gui", line 97, in __init__
  File "cosmonium.ui.gui", line 138, in calc_scale
TypeError: 'float' object is not subscriptable
Traceback (most recent call last):
  File "__main__", line 296, in <module>
  File "direct.showbase.ShowBase", line 3325, in run
  File "direct.task.Task", line 541, in run
  File "direct.task.Task", line 495, in step
  File "cosmonium.workers", line 66, in callbackTask
  File "cosmonium.cosmonium", line 462, in configure_scene
  File "cosmonium.ui.gui", line 97, in __init__
  File "cosmonium.ui.gui", line 138, in calc_scale
TypeError: 'float' object is not subscriptable
```

**Comment by el-dee** [comment]
Thank you for reporting these issues, that's indeed a pretty large eclipse :smiling_face_with_tear: 

I believe this was fixed in dev build [v0.3.0.dev837](https://github.com/cosmonium/cosmonium/releases/tag/v0.3.0.dev837) though there is still an issue : the shadow is not rendered if the Moon hasn't been rendered once first. This will be fixed in the next dev build.

Also the log files has been moved at their correct places in v0.3.0, you can find it now at `~/.cache/cosmonium/log/output.log` (The wiki page was still referring to the old location, I fixed that too)

Rollbacking to V0.2.2 is I'm afraid not possible, unless you remove the configuration file as some parameters have been updated and are no longer compatible when reverting the version.

![screenshot-2024-04-09-21-33-26-395](https://github.com/cosmonium/cosmonium/assets/44778133/2c5ee2c8-465a-473c-84a8-572a7444afb2)

**Comment by C1ffisme** [comment]
Unfortunately said version is the one I happen to be using. However, now that I know the location of the logfile, I think the logs are rather revealing:

```
Using C++ Engine
ERROR: Could not load cefpanda
Traceback (most recent call last):
  File "cosmonium.ui.windows.browser", line 28, in <module>
  File "importlib._bootstrap", line 1007, in _find_and_load
  File "importlib._bootstrap", line 986, in _find_and_load_unlocked
  File "importlib._bootstrap", line 680, in _load_unlocked
  File "importlib._bootstrap", line 838, in exec_module
  File "cefpanda", line 7, in <module>
  File "importlib._bootstrap", line 1007, in _find_and_load
  File "importlib._bootstrap", line 986, in _find_and_load_unlocked
  File "importlib._bootstrap", line 680, in _load_unlocked
  File "importlib._bootstrap", line 838, in exec_module
  File "cefpython3", line 64, in <module>
Exception: Python version not supported: 3.9.13 (main, Aug 15 2022, 06:11:49) 
[GCC 10.2.1 20210130 (Red Hat 10.2.1-11)]
WARNING: Token 'BOOL' defined, but not used
WARNING: Token 'FLOAT' defined, but not used
WARNING: Token 'INT' defined, but not used
WARNING: Token 'NAME' defined, but not used
WARNING: There are 4 unused tokens
Loading config file /home/c1ff/.config/cosmonium/config.yaml
Found languages: en_US.UTF-8
Cosmonium version: V0.3.0.dev837
Python version: 3.9.13
Panda version: 1.11.0 (d9052bae763ac42d0bf1d678be7206b20f7e46f8) by cosmonium (Mar 15 2024 20:58:46)
Panda Systems:
	 Bullet
	 Freetype
	 OpenSSL
	 eigen
	 libjpeg
	 libpng
	 libtiff
	 nativenet
	 net
	 openexr
	 pipelining
	 stdfloat-double
	 system
	 threads
	 zlib
Data type: double
Panda config file /home/c1ff/.config/cosmonium/config.prc
:audio(error):   load_dso(libp3openal_audio.so) failed, will use NullAudioManager
:audio(error):     No error.
Known pipe types:
  glxGraphicsPipe
(all display modules loaded.)
ATTENTION: default value of option vblank_mode overridden by environment.
Hardware Vendor: Intel
Driver Renderer: Mesa Intel(R) HD Graphics 530 (SKL GT2) (4.6 (Compatibility Profile) Mesa 24.0.0-1pop0~1706872735~22.04~0fa430c)
Shader version: 460
sRGB framebuffer: False
sRGB textures: True
Hardware instancing: True
Hardware tessellation: True
Inverse-Z buffer: False
Render to buffer: True
Floating point buffer: True
Texture array: True
Cache directory: /home/c1ff/.cache/cosmonium
Loading /home/c1ff/Applications/cosmonium/data/defaults.yaml (cached)
WARNING: Missing period
WARNING: Missing period
Loading /home/c1ff/Applications/cosmonium/data/cosmonium.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/solar-system.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/frames.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/ssd.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/manual-orbits.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/sun/sun.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/mercury/mercury.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/venus/venus.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/earth/earth.yaml (cached)
Add scattering on atmosphere
Loading /home/c1ff/Applications/cosmonium/data/solar-system/earth/moon/moon.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/earth/hst/hst.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/earth/iss/iss.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/mars/mars.yaml (cached)
Add scattering on atmosphere
Loading /home/c1ff/Applications/cosmonium/data/solar-system/jupiter/jupiter.yaml (cached)
Add scattering on atmosphere
Loading /home/c1ff/Applications/cosmonium/data/solar-system/jupiter/inner-moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/jupiter/moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/jupiter/irregular-moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/saturn/saturn.yaml (cached)
Add scattering on atmosphere
Loading /home/c1ff/Applications/cosmonium/data/solar-system/saturn/inner-moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/saturn/moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/saturn/irregular-moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/uranus/uranus.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/uranus/inner-moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/uranus/moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/uranus/irregular-moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/neptune/neptune.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/neptune/inner-moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/neptune/moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/neptune/irregular-moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/dwarfs/dwarfs.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/dwarfs/pluto/pluto.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/dwarfs/pluto/moons.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/dwarfs/ceres/ceres.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/dwarfs/eris/eris.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/dwarfs/haumea/haumea.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/dwarfs/makemake/makemake.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/asteroids/asteroids.yaml (cached)
DB rotations : Element astraea not found
DB rotations : Element 6-hebe not found
DB rotations : Element 7-iris not found
DB rotations : Element 8-flora not found
DB rotations : Element 9-metis not found
DB rotations : Element 10-hygiea not found
Loading /home/c1ff/Applications/cosmonium/data/solar-system/tnos/tnos.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/comets/comets.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/solar-system/interstellars/interstellars.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/stars.yaml (cached)
Unknown class '(K)'
Unknown class 'esdK7'
Unknown class 'esdM1.5'
Loading /home/c1ff/Applications/cosmonium/data/galaxies.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/procedural/procedural.yaml (cached)
Add scattering on atmosphere
Unknown appearance type 'textures-dict' {'type': 'textures-dict', 'entries': {'grass': 'textures/grass/GrassGreenTexture0003.jpg', 'snow': 'textures/snow/SnowIceTexture0007.jpg', 'rock': 'textures/rock/SeamlessMixedStone0009.jpg', 'gravel': 'textures/gravel/gravel_01.png', 'soil': 'textures/soil/dirt_02.png'}, 'scale': [0.5, 0.5], 'tiling': 'hash'}
Unknown appearance type 'textures-dict' {'type': 'textures-dict', 'entries': {'ash': {'albedo': 'ash/Volcanic_Ash_001_COLOR.jpg', 'normal': 'ash/Volcanic_Ash_001_NORM.jpg'}}, 'scale': [0.5, 0.5], 'tiling': 'hash'}
Creating system for Gas Demo
Loading /home/c1ff/Applications/cosmonium/data/constellations.yaml (cached)
Loading /home/c1ff/Applications/cosmonium/data/asterisms.yaml (cached)
Could not find star ALF CRU
Building tree
Using xclip clipboard
Switching navigation controller to Free navigation
Switching ship to No ship
Switching camera to Fixed camera
Using scene manager 'region'
Ambient light level:  0.60
New nearest system: Sol System
New nearest visible object: Sun
Select Earth
Go to front Earth
Follow Earth
New nearest visible object: Earth
Add scattering on None
Add scattering on clouds
Leaving atmosphere
Creating shader basic-ut-sl-t (/home/c1ff/.cache/cosmonium/shaders/adb03710b5e355bb076089f67cbc7751)
Creating shader basic-ut-flat-pt-sta-ut (/home/c1ff/.cache/cosmonium/shaders/285fbef135bbc172ea1556f4b520e9fd)
Creating shader basic-ut-flat-ult (/home/c1ff/.cache/cosmonium/shaders/ff332122cba2500e2300d9515613c927)
:display:gsg:glgsg(error): An error occurred while compiling GLSL fragment shader /home/c1ff/.cache/cosmonium/shaders/285fbef135bbc172ea1556f4b520e9fd.fragment.glsl:
/home/c1ff/.cache/cosmonium/shaders/285fbef135bbc172ea1556f4b520e9fd.fragment.glsl:22(13): error: no function with name 'texture2D'
/home/c1ff/.cache/cosmonium/shaders/285fbef135bbc172ea1556f4b520e9fd.fragment.glsl:23(17): warning: `tex0' used uninitialized
/home/c1ff/.cache/cosmonium/shaders/285fbef135bbc172ea1556f4b520e9fd.fragment.glsl:28(15): warning: `tex0' used uninitialized
:display:gsg:glgsg(error): An error occurred while compiling GLSL fragment shader /home/c1ff/.cache/cosmonium/shaders/ff332122cba2500e2300d9515613c927.fragment.glsl:
/home/c1ff/.cache/cosmonium/shaders/ff332122cba2500e2300d9515613c927.fragment.glsl:22(13): error: no function with name 'texture2D'
/home/c1ff/.cache/cosmonium/shaders/ff332122cba2500e2300d9515613c927.fragment.glsl:28(15): warning: `tex0' used uninitialized
New buffer float_color color_bits=96 red_bits=32 green_bits=32 blue_bits=32  size=(256, 256) 
Creating shader oneil (/home/c1ff/.cache/cosmonium/shaders/f408405ed7964bb720909da95e761a98)
Creating shader basic-uensi-lambert-ls-oneil-ground-infrag-hdr-norm--munser-gl (/home/c1ff/.cache/cosmonium/shaders/d901350d7bc2f312d7145c22be31b935)
Creating shader basic-ut-lambert-ls-oneil-ground-infrag-hdr-norm--mutr-gl (/home/c1ff/.cache/cosmonium/shaders/358f28f4cdc7bfcf9fb0ec5827ab64c0)
Creating shader basic-u-oneil-sky-infrag-hdr-norm-mr-gl (/home/c1ff/.cache/cosmonium/shaders/7c236856bfe4cb622b3178dd511b90ab)
Creating shader basic-ut-pt-galaxy-ut-ncp (/home/c1ff/.cache/cosmonium/shaders/30c46aac24d988d0b8f8eea4fcbc09a7)
Unknown source 'global_lights' requested
:display:gsg:glgsg(error): An error occurred while compiling GLSL fragment shader /home/c1ff/.cache/cosmonium/shaders/30c46aac24d988d0b8f8eea4fcbc09a7.fragment.glsl:
/home/c1ff/.cache/cosmonium/shaders/30c46aac24d988d0b8f8eea4fcbc09a7.fragment.glsl:21(13): error: no function with name 'texture2D'
/home/c1ff/.cache/cosmonium/shaders/30c46aac24d988d0b8f8eea4fcbc09a7.fragment.glsl:22(17): warning: `tex0' used uninitialized
/home/c1ff/.cache/cosmonium/shaders/30c46aac24d988d0b8f8eea4fcbc09a7.fragment.glsl:29(15): warning: `tex0' used uninitialized
:display:gsg:glgsg(error): An error occurred while compiling GLSL fragment shader /home/c1ff/.cache/cosmonium/shaders/7c236856bfe4cb622b3178dd511b90ab.fragment.glsl:
/home/c1ff/.cache/cosmonium/shaders/7c236856bfe4cb622b3178dd511b90ab.fragment.glsl:101(17): error: no function with name 'texture2D'
/home/c1ff/.cache/cosmonium/shaders/7c236856bfe4cb622b3178dd511b90ab.fragment.glsl:119(18): error: no function with name 'texture2D'
/home/c1ff/.cache/cosmonium/shaders/7c236856bfe4cb622b3178dd511b90ab.fragment.glsl:126(18): error: no function with name 'texture2D'
:display:gsg:glgsg(error): An error occurred while compiling GLSL fragment shader /home/c1ff/.cache/cosmonium/shaders/d901350d7bc2f312d7145c22be31b935.fragment.glsl:
/home/c1ff/.cache/cosmonium/shaders/d901350d7bc2f312d7145c22be31b935.fragment.glsl:163(17): error: no function with name 'texture2D'
/home/c1ff/.cache/cosmonium/shaders/d901350d7bc2f312d7145c22be31b935.fragment.glsl:181(18): error: no function with name 'texture2D'
/home/c1ff/.cache/cosmonium/shaders/d901350d7bc2f312d7145c22be31b935.fragment.glsl:188(18): error: no function with name 'texture2D'
/home/c1ff/.cache/cosmonium/shaders/d901350d7bc2f312d7145c22be31b935.fragment.glsl:237(13): error: no function with name 'texture2D'
/home/c1ff/.cache/cosmonium/shaders/d901350d7bc2f312d7145c22be31b935.fragment.glsl:238(13): error: no function with name 'texture2D'
/home/c1ff/.cache/cosmonium/shaders/d901350d7bc2f312d7145c22be31b935.fragment.glsl:239(13): error: no function with name 'texture2D'
/home/c1ff/.cache/cosmonium/shaders/d901350d7bc2f312d7145c22be31b935.fragment.glsl:240(13): error: no function with name 'texture2D'
/home/c1ff/.cache/cosmonium/shaders/d901350d7bc2f312d7145c22be31b935.fragment.glsl:241(22): warning: `tex0' used uninitialized
/home/c1ff/.cache/cosmonium/shaders/d901350d7bc2f312d7145c22be31b935.fragment.glsl:242(18): warning: `tex3' used uninitialized
/home/c1ff/.cache/cosmonium/shaders/d901350d7bc2f312d7145c22be31b935.fragment.glsl:243(22): warning: `tex1' used uninitialized
/home/c1ff/.cache/cosmonium/shaders/d901350d7bc2f312d7145c22be31b935.fragment.glsl:245(18): warning: `tex2' used uninitialized
:display:gsg:glgsg(error): An error occurred while compiling GLSL fragment shader /home/c1ff/.cache/cosmonium/shaders/358f28f4cdc7bfcf9fb0ec5827ab64c0.fragment.glsl:
/home/c1ff/.cache/cosmonium/shaders/358f28f4cdc7bfcf9fb0ec5827ab64c0.fragment.glsl:148(17): error: no function with name 'texture2D'
/home/c1ff/.cache/cosmonium/shaders/358f28f4cdc7bfcf9fb0ec5827ab64c0.fragment.glsl:166(18): error: no function with name 'texture2D'
/home/c1ff/.cache/cosmonium/shaders/358f28f4cdc7bfcf9fb0ec5827ab64c0.fragment.glsl:173(18): error: no function with name 'texture2D'
/home/c1ff/.cache/cosmonium/shaders/358f28f4cdc7bfcf9fb0ec5827ab64c0.fragment.glsl:212(13): error: no function with name 'texture2D'
/home/c1ff/.cache/cosmonium/shaders/358f28f4cdc7bfcf9fb0ec5827ab64c0.fragment.glsl:213(17): warning: `tex0' used uninitialized
New nearest visible object: ISS
New nearest visible object: Earth
New nearest visible object: ISS
New nearest visible object: Earth
```

**Comment by el-dee** [comment]
With the complete log, it's indeed a completely different problem !
Cosmonium is using OpenGL Compatibility Profile, so that the shaders can easily be used on OpenGL 2 and OpenGL 3+ hardware. The function `texture2D()` is provided by this compatibility profile, amongst others.
Now it seems that this version of the Intel driver does not have this function available even in Compatibility Profile, which is pretty weird.

**Comment by el-dee** [comment]
This should be fixed in 6f22098bf3b2876e61a6ec99aaadbc8c7a63d8b4 (and support for older OpenGL profile has been cleaned up in subsequent commits), I must still test this on other hardware to be sure I didn't introduced any regression.

A new dev build will be made shortly.
